kratos/solving_strategies/convergencecriterias/mixed_generic_criteria.h

```diff
@@ -454,6 +506,41 @@ class MixedGenericCriteria : public ConvergenceCriteria< TSparseSpace, TDenseSpa
         }
     }
 
+    /**
+     * @brief This method generates the list of variables from Parameters
+     * @param ThisParameters Input parameters
+     * @return List of variables considered as input
+     */
+    ConvergenceVariableListType GenerateConvergenceVariableListFromParameters(Kratos::Parameters ThisParameters)
+    {
+      // Iterate over variables
+      ConvergenceVariableListType aux_list;
+      for (auto& r_param : ThisParameters["convergence_variables_list"]) {
+          const std::string& r_variable_name = r_param["variable"].GetString();
+
+          // Variable point
+          VariableData* p_variable = nullptr;
+
+          // Options are double or array
+          if (KratosComponents<Variable<double>>::Has(r_variable_name)) {
+              const Variable<double>& r_variable = KratosComponents<Variable<double>>::Get(r_variable_name);
+              p_variable = &const_cast<Variable<double>&>(r_variable);
```

Comment:
I think this was not working because `typedef std::vector<std::tuple<VariableData*, TDataType, TDataType>> ConvergenceVariableListType;` should be `typedef std::vector<std::tuple<const VariableData*, TDataType, TDataType>> ConvergenceVariableListType;` I would say
can you try please?